arrow-core-data/src/main/kotlin/arrow/typeclasses/Alternative.kt

```diff
@@ -80,11 +87,17 @@ fun <T, F, A> Kind<T, Kind<F, A>>.altSum(AF: Alternative<F>, FT: Foldable<T>): K
   }
 }
 
-fun <T, F, A> Kind<T, A>.altFold(AF: Alternative<F>, FT: Foldable<T>): Kind<F, A> = FT.run { toList().altFromList(AF) }
+fun <T, F, A> Kind<T, A>.altFold(AF: Alternative<F>, FT: Foldable<T>): Kind<F, A> =
+  FT.run { toList().altFromList(AF) }
 
-fun <F, A> List<A>.altFromList(AF: Alternative<F>): Kind<F, A> = map { AF.just(it) }.k().altSum(AF, object : Foldable<ForListK> {
-  override fun <A, B> Kind<ForListK, A>.foldLeft(b: B, f: (B, A) -> B): B = fix().foldLeft(b, f)
-  override fun <A, B> Kind<ForListK, A>.foldRight(lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> = fix().foldRight(lb, f)
-})
+fun <F, A> List<A>.altFromList(AF: Alternative<F>): Kind<F, A> =
+  map { AF.just(it) }.k().altSum(AF, object : Foldable<ForListK> {
```

Comment:
Can we not use `ListK.foldable()` here directly? Instead of re-creating the `Foldable<ForListK>` object every time we call this function?

Reply:
Not from `arrow-core-data` because `ListK.foldable` is in `arrow-core`. I wrote this and was very annoyed at that as well :/

Reply:
Oh right, yes I've been there before :D Can we just keep a locally cached value then instead?